AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/holder/EBeanHolder.java

```diff
@@ -111,23 +116,30 @@ public void invokeInitInConstructors() {
 		overloadedConstructorBody.invoke(getInit());
 	}
 
-	public void createFactoryMethod(boolean hasSingletonScope) {
+	public void createFactoryMethod(EBean.Scope scope) {
 
 		AbstractJClass narrowedGeneratedClass = codeModelHelper.narrowGeneratedClass(generatedClass, annotatedElement.asType());
 
 		JMethod factoryMethod = generatedClass.method(PUBLIC | STATIC, narrowedGeneratedClass, GET_INSTANCE_METHOD_NAME);
-
 		codeModelHelper.generify(factoryMethod, annotatedElement);
 
 		JVar factoryMethodContextParam = factoryMethod.param(getClasses().CONTEXT, "context");
 
 		JBlock factoryMethodBody = factoryMethod.body();
 
-		/*
-		 * Singletons are bound to the application context
-		 */
-		if (hasSingletonScope) {
+		switch (scope) {
+		case Default:
+		case Fragment:
```

Comment:
How come `Default` and `Fragment` is in the same branch? Maybe i do not understand something?

Reply:
Because they share common code, but note that below it is called `createOverloadedFactoryMethod`, and it is there where a difference between both scopes is set. The fragment scope code is:

```java
    public static TestModel_ getInstance_(Context context, Object rootFragment) {
        if (rootFragment instanceof BeanHolder) {
            BeanHolder beanHolder = ((BeanHolder) rootFragment);
            TestModel_ bean = beanHolder.getBean(TestModel_.class);
            if (bean == null) {
                bean = new TestModel_(context, rootFragment);
                beanHolder.putBean(TestModel_.class, bean);
            }
            return bean;
        }
        return new TestModel_(context, rootFragment);
    }
```

The Default scope only has:

```java
    public static TestModel_ getInstance_(Context context, Object rootFragment) {
        return new TestModel_(context, rootFragment);
    }
```